mubench.pipeline/data/detector_execution.py

```diff
@@ -69,13 +65,11 @@ def __init__(self, run_mode: DetectorMode, detector: Detector, version: ProjectV
         self.findings_filter = findings_filter
 
     def execute(self, compile_base_path: str, timeout: Optional[int], logger: Logger):
-        detector_invocation = ["java"] + self.detector.java_options + ["-jar", _quote(self.detector.jar_path)]
-        command = detector_invocation + self._get_detector_arguments(self.version.get_compile(compile_base_path))
-        command = " ".join(command)
+        detector_args = self._get_detector_arguments(self.version.get_compile(compile_base_path))
 
         start = time.time()
         try:
-            Shell.exec(command, logger=logger, timeout=timeout)
+            self.detector.runner_interface.execute(self.version, detector_args, timeout, logger)
```

Comment:
Valid point. I think I would like the task to look like this

    try:
      self.detector.execute(...)  # internally delegates to the interface
    except NoCompatibleRunnerInterface as e:
      self.logger.error("Cannot run detector, because it has no compatible runner interface", e)
      return self.skip(...)
    except ...
      ...

instead of introducing a second level of error handling through an if. This we could realize by a `NoInterface` implementation of `RunnerInterface`, which raises the respective exception on execute ;)